src/api/eventTypes.js

```diff
@@ -173,3 +175,253 @@ export type RestartEvent = $ReadOnly<{|
   zulip_version?: string,
   zulip_feature_level?: number,
 |}>;
+
+/* eslint-disable flowtype/generic-spacing */
+
+/**
+ * Data updates we might get in `update` or `update_dict` realm events.
+ *
+ * We expect this to be live-updates to the data we request by putting
+ * 'realm' in `fetch_event_types` when we call POST /register. We type the
+ * shape of that data as InitialDataRealm. The properties are named
+ * differently between here and there (InitialDataRealm has lots of
+ * properties that start with "realm_"), but we expect the values to be the
+ * same types.
+ */
+// If there's a property that exists in InitialDataRealm, but we don't use
+// it and the event doc doesn't mention it, write the InitialDataRealm
+// property in a code comment in the place where it would go here.
+/* prettier-ignore */
+type RealmDataForUpdate = $ReadOnly<{
+  //
+  // Keep alphabetical by the InitialDataRealm property. So by
+  // realm_allow_edit_history, not by allow_edit_history, etc.
+  //
+
+  development_environment:
+    $PropertyType<InitialDataRealm, 'development_environment'>,
+  // <insert_name>:
+  //   $PropertyType<InitialDataRealm, 'event_queue_longpoll_timeout_seconds'>,
+  jitsi_server_url:
+    $PropertyType<InitialDataRealm, 'jitsi_server_url'>,
+  max_avatar_file_size_mib:
+    $PropertyType<InitialDataRealm, 'max_avatar_file_size_mib'>,
+  max_file_upload_size_mib:
+    $PropertyType<InitialDataRealm, 'max_file_upload_size_mib'>,
+
+  // TODO: Check if the doc really means this, instead of max_icon_file_size
+  //   and max_icon_file_size_mib?
+  icon_file_size: number,
+
+  // <insert_name>:
+  //   $PropertyType<InitialDataRealm, 'max_logo_file_size'>,
+  // <insert_name>:
+  //   $PropertyType<InitialDataRealm, 'max_logo_file_size_mib'>,
+  password_min_length:
+    $PropertyType<InitialDataRealm, 'password_min_length'>,
```

Comment:
It looks like max_message_length and its neighbors didn't make it into this list.